### PR TITLE
Include unit and integration test

### PR DESCRIPTION
**Pull request by Abuelodelanada**
## Issue

This PR includes unit and integration test

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -4,72 +4,89 @@
...
-        # Trigger a config-changed event with an updated value
-        self.harness.update_config({"log-level": "debug"})
-        # Check the charm is in WaitingStatus
-        self.assertIsInstance(self.harness.model.unit.status, WaitingStatus)
+    @patch("charm.MimirK8SOperatorCharm._mimir_version", new_callable=PropertyMock)
+    def test_mimir_pebble_ready_cannot_connect(self, mock_version):
+        mock_version.return_value = "2.4.0"
+        ops.testing.SIMULATE_CAN_CONNECT = False
+        self.harness.container_pebble_ready("mimir")
+        self.assertEqual(self.harness.model.unit.status, WaitingStatus("Waiting for Pebble ready"))
```
Comment (sed-i) [inline tests/unit/test_charm.py:73]: I'm confused: can we keep `SIMULATE_CAN_CONNECT` as True and just not emit pebble-ready?

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -4,72 +4,89 @@
...
+
+    @patch.object(Container, "push")
+    @patch("charm.MimirK8SOperatorCharm._current_mimir_config", new_callable=PropertyMock)
+    @patch("charm.MimirK8SOperatorCharm._set_alerts", new_callable=Mock)
+    @patch("charm.MimirK8SOperatorCharm._mimir_version", new_callable=PropertyMock)
+    def test_mimir_pebble_ready_cannot_push_config(
+        self, mock_version, mock_set_alerts, mock_current_mimir_config, mock_push
+    ):
+        mock_version.return_value = "2.4.0"
+        mock_set_alerts.return_value = True
```
Comment (sed-i) [inline tests/unit/test_charm.py:83]: How about `patcher.start()` in setUp():
https://github.com/canonical/prometheus-k8s-operator/blob/7f9c38c6b712c8a4060919c7f46acba9a9ca2117/tests/unit/test_charm_status.py#L45

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -4,72 +4,89 @@
...
-from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from ops.model import ActiveStatus, BlockedStatus, Container, WaitingStatus
+from ops.pebble import Error as PebbleError
+from ops.pebble import PathError
 from ops.testing import Harness
 
 from charm import MimirK8SOperatorCharm
 
 
 class TestCharm(unittest.TestCase):
```
Comment (sed-i) [inline tests/unit/test_charm.py:18]: I wonder if should move away from `test_charm` and `TestCharm` in favor of more scoped naming, for example:
- `test_pebble_layer.py`
  - `class TestPebbleLayer`
    - `def test_in_isolation()`
      - assert service is running
      - assert pebble command args
    - `def test_with_external_url()`
      - assert arg has external url
- `test_charm_status.py`
  - `class TestPebbleErrors`
    - `def test_pebble_error_on_push_blocks_charm`

File: tox.ini [hunk 4]
```diff
@@ -4,13 +4,13 @@
 [tox]
 skipsdist=True
 skip_missing_interpreters = True
-envlist = lint, unit
+envlist = lint, static-{charm,lib}, unit
 
 [vars]
 src_path = {toxinidir}/src/
 tst_path = {toxinidir}/tests/
```
Comment (sed-i) [inline tox.ini:11]:
```suggestion
src_path = {toxinidir}/src
tst_path = {toxinidir}/tests
```

File: tox.ini [hunk 5]
```diff
@@ -52,6 +52,27 @@ commands =
...
+setenv =
+    unit: MYPYPATH = {[vars]tst_path}/unit
+    integration: MYPYPATH = {[vars]tst_path}/integration
+deps =
+    mypy
+    types-dataclasses
+    types-PyYAML
+    types-requests
+    charm: -r{toxinidir}/requirements.txt
+    lib: git+https://github.com/canonical/operator#egg=ops
```
Comment (sed-i) [inline tox.ini:66]: Just `ops` works now?

File: tests/unit/test_charm.py (hunk 1 above)
Comment (Abuelodelanada) [reply tests/unit/test_charm.py:73]: The name of the test is confusing... I'll change it.

Now you mention it, it's weird that you can emit pebble-ready after setting `SIMULATE_CAN_CONNECT = False`

File: tests/unit/test_charm.py (hunk 3 above)
Comment (Abuelodelanada) [reply tests/unit/test_charm.py:18]: I did not do it because the charm is now in its first steps, we can do ir though

**Pull request merged**